react/features/filmstrip/middleware.js

```diff
@@ -31,6 +34,23 @@ MiddlewareRegistry.register(({ getState }) => next => action => {
             return result;
         }
         break;
+
+    case SET_FILMSTRIP_VISIBLE:
+        // FIXME: Only do this on mobile for now. The logic for participant
+        // pinning / unpinning is not on React yet so dispatching the action
+        // is not enough.
+        if (typeof APP === 'undefined') {
+            const { audioOnly } = getState()['features/base/conference'];
+
+            if (!audioOnly) {
```

Comment:
Here I'm commenting on the commit "[RN] Unpin participant and set last N to 1 if the filmstrip is hidden" only:

I really don't understand the involvement of `audioOnly`. I guess `audioOnly` has an interaction with `setLastN`. But I totally don't see why it interacts with `pinParticipant`. The way I think of the subject is: if the filmstrip is disabled, then maybe pinning is something we don't want because it's hard to change it without the filmstrip at least on mobile (and that's unrelated to whether the filmstrip displays videos or not i.e. audio-only mode or not) and we don't want to receive more than we're going to display (and that's related to audio-only mode probably).

Reply:
Good point, I'll do that outside the if, unconditionally.